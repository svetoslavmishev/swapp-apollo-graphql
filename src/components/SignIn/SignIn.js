import React, { useState, useContext } from 'react';
import PropTypes from 'prop-types';
import { useMutation, useApolloClient } from '@apollo/react-hooks';
import {
  Container,
  FormControl,
  FormLabel,
  Input,
  Paper,
  Button
} from '@material-ui/core';

import { SIGN_IN } from '../../queries/queries';
import { ThemeContext } from '../../themeContext';
import styles from './SignInStyles';
import Loading from '../shared/Loading/Loading';

function SignIn({ history }) {
  const { currentTheme } = useContext(ThemeContext);
  const classes = styles({ currentTheme });
  const [fields, setFields] = useState({
    email: 'demo@st6.io',
    password: 'demo1234'
  });
  const client = useApolloClient();
  const [signIn, { loading, error }] = useMutation(SIGN_IN, {
    onCompleted: ({ signIn: { token } }) => {
      localStorage.setItem('token', token);
      client.writeData({ data: { isAuthenticated: true } });
      history.push('/episodes');
      setFields({ email: '', password: '' });
    },
    onError: e => {
      console.log(e);
    }
  });

  const handleChange = ({ target: { value, name } }) => {
    setFields({ ...fields, [name]: value });
  };

  const handleSubmit = ev => {
    ev.preventDefault();
    signIn({ variables: { ...fields } });
    setFields({ email: '', password: '' });
  };

  if (loading) return <Loading />;

  const hasError = error && error.message ? true : false;

  return (
    <Container className={classes.root} maxWidth="xl">
      <div className="md:w-1/3">
        <div className={`${classes.logo} text-center`}>
          {process.env.REACT_APP_TITLE}
        </div>
        <Paper className={classes.paper}>
          <FormControl error={hasError}>
            {hasError && (
              <FormLabel component="legend">{error.message}</FormLabel>
            )}
            <Input
              className={`${classes.inputs} mt-2 mb-2`}
              id="email-component-helper"
              type="text"
              name="email"
              value={fields.email}
              onChange={handleChange}
              aria-describedby="component-helper-text"
              placeholder="email"
            />
          </FormControl>
          <FormControl>
            <Input
              className={`${classes.inputs} mt-2 mb-2`}
              id="password-component-helper"
              type="password"
              name="password"
              value={fields.password}
              onChange={handleChange}
              aria-describedby="component-helper-text"
              placeholder="password"
            />
          </FormControl>
          <Button
            className={`${classes.solidButton} ${classes.button}`}
            variant="contained"
            onClick={handleSubmit}
            disabled={!fields.email && !fields.password}
          >
            Login
          </Button>
        </Paper>
      </div>
    </Container>
  );
}

SignIn.propTypes = {
  history: PropTypes.object
};

export default SignIn;
