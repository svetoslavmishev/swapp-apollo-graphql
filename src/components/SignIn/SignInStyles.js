import { makeStyles } from '@material-ui/styles';

const styles = makeStyles(theme => ({
  inputs: ({ currentTheme }) => theme.palette[currentTheme].Inputs,
  solidButton: ({ currentTheme }) => theme.palette[currentTheme].SolidButtons,
  progress: ({ currentTheme }) => ({
    background: theme.palette[currentTheme].backgroundColor
  })
}));

export default styles;
