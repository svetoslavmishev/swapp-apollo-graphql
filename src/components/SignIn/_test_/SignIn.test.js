import React from 'react';
import { shallow } from 'enzyme';
import { ThemeContext, theme } from '../../../themeContext';
import SignIn from '../SignIn';

describe('<SignIn />', () => {
  it('renders <SignIn /> without error', () => {
    const wrapper = shallow(
      <ThemeContext.Provider value={theme}>
        <SignIn />
      </ThemeContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
