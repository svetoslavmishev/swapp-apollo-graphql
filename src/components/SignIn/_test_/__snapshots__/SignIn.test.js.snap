// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> renders <SignIn /> without error 1`] = `ShallowWrapper {}`;
